### PR TITLE
Module8: Useful libraries

### PR DESCRIPTION
**Pull request by nkorczyk**
Changes proposed in this PR:

Added `Eslint` configuration.
Introduced `flow`.
Introduced `styled-components`.
Introduced `reselect`.

Please check and review.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,14 +1,41 @@
...
+      2
+    ],
+    "import/no-extraneous-dependencies": [
+      "error",
+      {
+        "devDependencies": true
+      }
+    ],
+    "global-require": 0,
+    "no-console": "off",
```
Comment (MichalWierzba) [inline .eslintrc]: I'd suggest leaving this rule enabled for client app (remove all console statements), and disabled for server side (it's ok to have console statement on server) - https://eslint.org/docs/rules/no-console

File: src/index.test.js [hunk 2]
```diff
@@ -1,3 +1,3 @@
-test("", () => {
-  expect("")
+test('', () => {
+  expect('');
```
Comment (MichalWierzba) [inline src/index.test.js]: I think this file should be removed ;)

File: src/serverRenderer.js [hunk 3]
```diff
@@ -24,7 +24,9 @@ function renderHTML(html, preloadedState) {
           <meta charset=utf-8>
           <title>React Server Side Rendering</title>
           <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/materialize/1.0.0/css/materialize.min.css"></link>
-          ${process.env.NODE_ENV === 'development' ? '' : '<link href="/css/main.css" rel="stylesheet" type="text/css">'}
+          ${process.env.NODE_ENV === 'development'
+      ? ''
+      : '<link href="/css/main.css" rel="stylesheet" type="text/css">'}
```
Comment (MichalWierzba) [inline src/serverRenderer.js]: Looks like there is an issue with indentation here.

File: package.json [hunk 4]
```diff
@@ -6,9 +6,12 @@
   "main": "script.js",
   "scripts": {
     "start": "cross-env NODE_ENV=development  node ./buildScripts",
-    "build": "cross-env NODE_ENV=production webpack --config ./webpack",
+    "build:prod": "cross-env NODE_ENV=production webpack --config ./webpack",
     "prod": "cross-env NODE_ENV=production  node ./buildScripts",
     "lint": "eslint ./ --cache --ignore-pattern .gitignore --fix",
+    "build": "babel src/ -d lib/",
```
Comment (MichalWierzba) [inline package.json]: What is the purpose of this script?

**Review by MichalWierzba — APPROVED** [review]
Looks good to me! 👍 

Please check minor comments/questions.

File: .eslintrc (hunk 1 above)
Comment (nkorczyk) [reply .eslintrc]: Thanks for hint. Fixed.

File: src/index.test.js (hunk 2 above)
Comment (nkorczyk) [reply src/index.test.js]: Removed.

File: src/serverRenderer.js (hunk 3 above)
Comment (nkorczyk) [reply src/serverRenderer.js]: I'm not sure about that, but indentation was automatically set by VSCode.
The code above 28 line and below 29 line is a `string` and this code is a `regular` code. So I think that's why indentation is different.
But for better look and feel, I indented those two lines manually.

File: package.json (hunk 4 above)
Comment (nkorczyk) [reply package.json]: I added this because it has been shown in mentoring movie but it looks like it is not useful here. Removed.

**Comment by MichalWierzba** [comment]
Reselect looks good :+1:

What do you think about adding few sample tests for selectors?

**Comment by nkorczyk** [comment]
Added test for selector & fixed linting errors.

**Comment by MichalWierzba** [comment]
@nkorczyk Thanks for all updates. 

Looks good to me, please merge :)

**Pull request merged**